packages/scenario/test/instance-edit.test.ts

```diff
@@ -0,0 +1,383 @@
+
+			/**
+			 * This condenses the basic assertions of both edit-specific semantics:
+			 *
+			 * - before `instanceID` -> after `deprecatedID`
+			 * - recomputed `preload="uid"`  -> after `instanceID`
+			 *
+			 * Note: assertion at depth 2 in a chain of edits effectively exercises
+			 * overwrite of the previous edit's `deprecatedID` value, rather than
+			 * naively appending multiple `deprecatedID` elements.
+			 */
+			const chainingConditions: readonly ChainingAssertion[] = [
+				(source, edited) => {
+					expect(edited).toHaveDeprecatedIDFromSource({
+						metaNamespaceURI,
+						sourceScenario: source,
+					});
+				},
+
+				(source, edited) => {
+					expect(edited).toHaveEditedPreloadInstanceID({
+						metaNamespaceURI,
+						sourceScenario: source,
+					});
+				},
+			];
+
+			const assertChained = (source: Scenario, edited: Scenario) => {
+				for (const assertCondition of chainingConditions) {
+					assertCondition(source, edited);
+				}
+			};
+
+			/**
+			 * Tests that the chaining exercised by {@link assertChained} does not
+			 * apply to instances {@link a} and {@link b}, in either ordering, by
+			 * asserting that at least one of the chaining conditions fails.
+			 */
+			const assertNotChained = (a: Scenario, b: Scenario) => {
+				const orderings = [
+					[a, b],
+					[b, a],
+				] as const;
+
+				let assertionFailure: unknown = null;
+
+				for (const ordering of orderings) {
+					const [source, edited] = ordering;
+
+					for (const assertCondition of chainingConditions) {
+						if (assertionFailure != null) {
+							break;
+						}
+
+						try {
+							assertCondition(source, edited);
+						} catch (error) {
+							assertionFailure = error;
+						}
+					}
+				}
+
+				expect(assertionFailure).toBeInstanceOf(Error);
+			};
+
+			// Typical pattern: chained sequential edits form a linked list
+			it('creates a chain of edited deprecatedID -> source instanceID references over subsequent edits', () => {
+				const { source, edit1, edit2 } = scenarios;
+
+				// Prerequisite
+				assertChained(source, edit1);
+
+				// Sanity/meaningfulness of test: edit2 is not chained from source
+				assertNotChained(source, edit2);
+
+				// Assert: edit2 is chained from edit1
+				assertChained(edit1, edit2);
+			});
+
+			// Spec design/intent: chained branched edits form a tree
+			it('creates a tree of edited deprecatedID -> source instanceID references over subsequent edits of a common ancestor instance', async () => {
+				const { source, edit1: branch1, edit2: leaf1 } = scenarios;
+
+				const branch2 = await source.proposed_editCurrentInstanceState();
+				const leaf2 = await branch2.proposed_editCurrentInstanceState();
+
+				assertChained(source, branch1);
+				assertChained(source, branch2);
+				assertChained(branch1, leaf1);
+				assertChained(branch2, leaf2);
+
+				assertNotChained(branch1, leaf2);
+				assertNotChained(branch2, leaf1);
+			});
```

Comment:
When I first wrote up this PR, I noticed I had forgotten to test that we don't append multiple `deprecatedID` elements. As I went to test that, I realized that what was really lacking was this pair of spec semantics:

- sequenced edits produces a linked list
- concurrent edits produces a tree

I'm pretty sure this was discussed in a team meeting, around the time we started queueing up edit feature work. I hope these two tests clarify those cases!